api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -189,6 +190,17 @@
    */
   void end();
 
+  /**
+   * Marks the end of {@code Span} execution with an explicit end {@link Timestamp}.
+   *
+   * <p>Only the timing of the first end call for a given {@code Span} will be recorded, and
+   * implementations are free to ignore all further calls.
+   *
+   * @param endTimestamp the explicit end {@link Timestamp} for this {@code Span}.
+   * @since 0.1.0
+   */
+  void end(Timestamp endTimestamp);
```

Comment:
To avoid breaking people in the future, should we go with an option pattern here? That will let us add more properties in the future without changing this.

Not 100% convinced but want to hear options.

Reply:
Sure, going with an option pattern sounds good to me. What about `EndOptions`, similar to what OC has? ;)